staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -140,6 +151,9 @@ func tokenSource(isCmd bool, gcpConfig map[string]string) (oauth2.TokenSource, e
 		if gcpConfig["scopes"] != "" {
 			return nil, fmt.Errorf("scopes can only be used when kubectl is using a gcp service account key")
 		}
+		if gcpConfig["credentials-file"] != "" {
```

Comment:
It's prior art, but you probably want to pull this out of the `isCmd` block; callers won't set `cmd-path` and `credentials-file` at the same time.  While you're at it, you can simplify this by removing the `isCmd` parameter and the `isCmdTokenSource` helper, as it is redundant.

Reply:
Thanks @jlowdermilk 
gotcha on removing the credentails-file check as for the other part of your comment are you suggesting I do something like this?
```golang
func tokenSource(gcpConfig map[string]string) (oauth2.TokenSource, error) {
 	// Command-based token source
 	if cmd := isCmdTokenSource(gcpConfig); len(cmd) != 0 {
                // check gcp cmd config here
        ...
```

Reply:
Yes, but without the helper

```
func tokenSource(gcpConfig map[string]string) (oauth2.TokenSource, error) {
 	// Command-based token source
 	if cmd, ok := gcpConfig["cmd-path"]; ok {
                // check gcp cmd config here
        ...
```

`isCmdTokenSource` can be removed entirely, it's equivalent to map key lookup.